ctmds/commodity_price_generator.py

```diff
@@ -0,0 +1,156 @@
+import numpy as np
+import pandas as pd
+
+from ctmds.constants import COMMODITY_BASE_PRICES
+from ctmds.country_datetime_series import get_country_datetime_series
+from ctmds.enums import Commodity
+from ctmds.enums import CountryCode
+from ctmds.enums import Granularity
+
+
+class CommodityPriceGenerator:
+    def __init__(self, commodity: Commodity):
+        self.commodity = Commodity[commodity]
+
+    def generate_price_series(
+        self,
+        for_date: str,
+        country_code: CountryCode,
+        granularity: Granularity = Granularity.HOURLY,
+        seed: int | None = None,
+    ) -> pd.Series:
+        """
+        Generate synthetic price series for the commodity.
+
+        Parameters
+        ----------
+        for_date : str
+            Date in 'YYYY-MM-DD' format.
+        country_code : CountryCode
+            Country code
+        granularity : Granularity, optional
+            Granularity of time. Default is 'hourly'.
+        seed : int | None, optional
+            Seed for random number generator. Default is None
+
+        Returns
+        -------
+        pd.Series
+            Series of prices with pd.DatetimeIndex index
+        """
+        dates = get_country_datetime_series(for_date, country_code, granularity)
+
+        match self.commodity:
```

Comment:
The `match` / `case` is nice. 

There's an even more "pythonic" way of doing this though. Because everything in Python is an object - including functions - you can have a dictionary that maps a commodity to the corresponding function like this:

    PRICE_FUNCTIONS={
        Commodity.CRUDE: generate_crude_prices,
        Commodity.NATGAS: generate_natgas_prices,
        Commodity.POWER: generate_power_prices,
    }
    
Then just call:

    PRICE_FUNCTIONS[self.commodity](dates, country_code, seed)